### ISSUE TITLE
--progress-bar not working

### ISSUE THREAD

**Issue by bm-silva** [genomehubs/goat-cli#22]
Hi, I am currently facing an issue while utilizing the goat-cli tool for a taxon search operation. Specifically, I have employed the following command:

`goat-cli taxon search -f organisms.txt -R species -abgkGBNDmpPSc --progress-bar`

The organisms.txt file comprises a list of 500 individual organisms. However, I have observed that the progress bar associated with the process does not show in my terminal screen (Ubuntu, WSL, Win 11). This leads me to wonder if there might be an error in my approach or configuration.

Any insights into the reason for the non-functional progress bar and guidance on rectifying this issue would be greatly appreciated.

**Comment by Euphrasiologist** [comment]
Yes the progress bar needs a bit of work, especially with parallel requests like the one you are making. Again, a copy of the organisms.txt file would be great for debugging. Thanks!

**Comment by Euphrasiologist** [comment]
So I've looked into this a bit more. The `--progress-bar` queries the request in progress, but with 500 concurrent requests, the mechanism by which I do this will have to be different. I haven't catered for this yet, which is why it currently shows as nothing. It's probably best I throw an error until it's implemented.
